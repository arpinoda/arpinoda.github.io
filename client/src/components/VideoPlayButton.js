import React from 'react';
import PropTypes from 'prop-types';

const VideoPlayButton = ({ onClick, style }) => {
  const svgStyle = {
    ...{
      height: '20%',
      top: '2%',
      right: '2%',
      cursor: 'pointer',
      filter: 'drop-shadow(0px 3px 2px rgba(0,0,0,0.2))',
    },
    ...style,
  };

  const iconStyle = { fill: '#666' };

  const circleStyle = { fill: 'rgba(255, 255, 255, 0.8)' };

  return (
    <svg
      onMouseUp={e => {
        e.preventDefault();
        onClick();
      }}
      style={svgStyle}
      className="absolute z3"
      viewBox="0 0 140 140"
    >
      <circle style={circleStyle} cx="70" cy="70" r="65" />
      <polygon
        id="shape"
        points="50,40 100,70 100,70 50,100, 50,40"
        style={iconStyle}
      />
    </svg>
  );
};

VideoPlayButton.propTypes = {
  onClick: PropTypes.func,
  style: PropTypes.object,
};

export default VideoPlayButton;
