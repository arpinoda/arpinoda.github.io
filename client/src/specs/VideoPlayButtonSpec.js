import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import VideoPlayButton from '../components/VideoPlayButton';

describe('<VideoPlayButton />', () => {
  test('Should not break with empty props', () => {
    const component = renderer.create(<VideoPlayButton />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should NOT contain prepended style', () => {
    const component = renderer.create(<VideoPlayButton />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should contain prepended style', () => {
    const component = renderer.create(
      <VideoPlayButton style={{ color: 'red' }} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should call onClick when clicked', () => {
    const onClick = jest.fn();

    const component = mount(<VideoPlayButton onClick={onClick} />);

    component.simulate('mouseup');

    expect(onClick.mock.calls.length).toBe(1);
  });
});
