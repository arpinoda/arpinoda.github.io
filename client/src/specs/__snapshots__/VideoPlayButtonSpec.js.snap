// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayButton /> Should NOT contain prepended style 1`] = `
<svg
  className="absolute z3"
  onMouseUp={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "filter": "drop-shadow(0px 3px 2px rgba(0,0,0,0.2))",
      "height": "20%",
      "right": "2%",
      "top": "2%",
    }
  }
  viewBox="0 0 140 140"
>
  <circle
    cx="70"
    cy="70"
    r="65"
    style={
      Object {
        "fill": "rgba(255, 255, 255, 0.8)",
      }
    }
  />
  <polygon
    id="shape"
    points="50,40 100,70 100,70 50,100, 50,40"
    style={
      Object {
        "fill": "#666",
      }
    }
  />
</svg>
`;

exports[`<VideoPlayButton /> Should contain prepended style 1`] = `
<svg
  className="absolute z3"
  onMouseUp={[Function]}
  style={
    Object {
      "color": "red",
      "cursor": "pointer",
      "filter": "drop-shadow(0px 3px 2px rgba(0,0,0,0.2))",
      "height": "20%",
      "right": "2%",
      "top": "2%",
    }
  }
  viewBox="0 0 140 140"
>
  <circle
    cx="70"
    cy="70"
    r="65"
    style={
      Object {
        "fill": "rgba(255, 255, 255, 0.8)",
      }
    }
  />
  <polygon
    id="shape"
    points="50,40 100,70 100,70 50,100, 50,40"
    style={
      Object {
        "fill": "#666",
      }
    }
  />
</svg>
`;

exports[`<VideoPlayButton /> Should not break with empty props 1`] = `
<svg
  className="absolute z3"
  onMouseUp={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "filter": "drop-shadow(0px 3px 2px rgba(0,0,0,0.2))",
      "height": "20%",
      "right": "2%",
      "top": "2%",
    }
  }
  viewBox="0 0 140 140"
>
  <circle
    cx="70"
    cy="70"
    r="65"
    style={
      Object {
        "fill": "rgba(255, 255, 255, 0.8)",
      }
    }
  />
  <polygon
    id="shape"
    points="50,40 100,70 100,70 50,100, 50,40"
    style={
      Object {
        "fill": "#666",
      }
    }
  />
</svg>
`;
